### PR TITLE
add parquet.SortFunc

### PR DESCRIPTION
**Pull request by achille-roussel**
This PR is a proposal for exposing new APIs to generate column sorting functions. The changes intend to address the comments I left on the original proposal from https://github.com/segmentio/parquet-go/pull/87

Let me know if you have any feedback!

### CODE REVIEW

**Comment by brancz** [comment]
I like this a lot. Happy to go exactly with this!

File: sort.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+// Apply applies options to c.
+func (c *SortConfig) Apply(options ...SortOption) {
+	for _, opt := range options {
+		opt.ConfigureSort(c)
+	}
+}
+
+// ConfigureSort satisfies the SortOption interface.
+func (c *SortConfig) ConfigureSort(config *SortConfig) {
```
Comment (kevinburkesegment) [inline sort.go:29]: I've made this comment before, but the ergonomics of having a value and then calling a function on it with another value of the same type, only to copy it, just feels very odd... 

Why do you want/need to pass it in? Can't you just have `c` pass a `SortConfig` to the caller?

Like I would maybe name this 

```go
func (c *SortConfig) Clone() *SortConfig {
    
}
```

Comment (achille-roussel) [reply sort.go:29]: As documented on the `SortConfig` type, the intent of this method is to allow passing the configuration as option to the constructor function, I'm not sure how we could do that with the `Clone` method, can you share more details of how you would do this?

Comment (kevinburkesegment) [reply sort.go:29]: To be honest I'd probably just make `SortFuncOf` accept a single argument, SortConfig, that's configured however people want things to be sorted...

**Comment by kevinburkesegment** [comment]
Agreed, I don't think it's a big thing and happy to just merge as is, but this is our chance to get the API right, so wanted to make a point of it.

File: sort.go (hunk 1 above)
Comment (achille-roussel) [reply sort.go:29]: Maybe we can open an issue to discuss the config management strategy we would prefer to adopt in the library, it's a broad topic with lots of material available both online and in internal code bases.

**Comment by kevinburkesegment** [comment]
Understood. I think the Option... style might make more sense when we're configuring more than just fields on a struct?

**Comment by achille-roussel** [comment]
> this is our chance to get the API right, so wanted to make a point of it.

For now I'm going to keep it consistent with the other configuration APIs, but I'll start a discussion about it in a Github issue.

**Pull request merged**